### ISSUE TITLE
Support newer Skia APIs

### ISSUE THREAD

**Issue by dirkschulze** [adobe/svg-native-viewer#19]
Post `chrome/m70` branch the code got refactored and the `include/` folder does not include all required headers anymore. So the current Skia port was only tested up to `chrome/m70`.

Recently, Skia is planning minor breakages to the API which may makes SVG Native Viewer not compile with latest Skia anymore.

Also, documentation should be added how to include Skia library for building.

**Comment by mpsuzuki** [comment]
I tested with chrome/m80, my minimum tweak (still some warning and notes are complained) is 90c0bf3ace54103f43a0c86ae3b2e7b0f1f3d595, the difference is very small. Main difference is the renaming the macro, which breaks the older name spaces.

**Comment by dirkschulze** [comment]
@mpsuzuki IMO only supporting the latest "stable release" of Skia is fine unless it is clear how to build it.

**Comment by dirkschulze** [comment]
We do have latest API support now.

**Comment by moazin** [comment]
@dirkschulze I think this issue needs to be reopened. A few things I observed as I tried to compile Skia with SNV on a Linux machine.

1. The C++ standard needs to be 14 in order for some Skia includes to work. (Change needs to be made to CMakeLists.txt)
2. Skia is supposed to be placed in `third_party` however this isn't documented in the Readme.
3. An include to `../third_party/skia/` as some includes are written in such a way.
4. `SkPath::kWinding_FillType` has been changed into `SkPathFillType::kWinding` and similarly `SkPath::kEvenOdd_FillType` has been changed into `SkFillType::kEvenOdd`.
5. `SkShader::TileMode` has been changed into `SkTileMode`.
6. The call to `mCanvas->drawImageRect` needs to have a `SkSamplingOptions` passed in as reference I think.
